sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -152,6 +153,17 @@ abstract class Optimizer(sessionCatalog: SessionCatalog, conf: SQLConf)
 }
 
 /**
+ * Remove useless DISTINCT for MAX and MIN.
+ * This rule should be applied before ReplaceDeduplicateWithAggregate.
+ */
+object EliminateDistinct extends Rule[LogicalPlan] {
+  override def apply(plan: LogicalPlan): LogicalPlan = plan transformExpressions  {
+    case AggregateExpression(max: Max, _, true, _) => AggregateExpression(max, Complete, false)
```

Comment:
The `ResolveFunctions` rule already set it to `Complete`. @gengliangwang we should just keep the AggregateMode unchanged.